src/hotspot/share/classfile/javaClasses.hpp

```diff
@@ -180,10 +181,24 @@ class java_lang_String : AllStatic {
     return h;
   }
 
+  static unsigned int hash_code(const char* utf8_str, size_t utf8_len) {
```

Comment:
It's inlined because the hash functions working on unicode are inlined just above it. For the full context:

https://github.com/openjdk/jdk/blob/d55552d5323ff8a4ec37c320199b650e63f230b4/src/hotspot/share/classfile/javaClasses.hpp?plain=1#L166-L197

I think these are simple enough to be ok inlined, whereas the oop-versions are more complex and modify the object, so it makes sense that they are not.